packages/wdio-browserstack-service/src/instrumentation/funnelInstrumentation.ts

```diff
@@ -142,3 +143,106 @@ function getReferrer(framework?: string) {
     const fullName = framework ? 'WebdriverIO-' + framework : 'WebdriverIO'
     return `${fullName}/${BSTACK_SERVICE_VERSION}`
 }
+
+const sendEvent = {
+    tcgDown: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestTcgDownResponse', config),
+    invalidTcgAuth: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestInvalidTcgAuthResponseWithUserImpact', config),
+    tcgAuthFailure: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestTcgAuthFailure', config),
+    tcgtInitSuccessful: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestTcgtInitSuccessful', config),
+    initFailed: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestInitFailedResponse', config),
+    tcgProxyFailure: (config: BrowserStackConfig) => fireFunnelTestEvent('SDKTestTcgProxyFailure', config),
+}
+
+function isProxyError(authResult: any): boolean {
+    return (authResult as BrowserstackHealing.InitErrorResponse).status === 502
```

Comment:
This might not be the right way to conclude if error is because of proxy.

Reply:
Yes, this needs to be checked. Added a comment earlier in v8

Reply:
This has been removed as discussed earlier. Resolving this conversation.